nimble/host/src/ble_uuid.c

```diff
@@ -74,8 +65,8 @@ ble_uuid_init_from_buf(ble_uuid_any_t *uuid, const void *buf, size_t len)
 int
 ble_uuid_cmp(const ble_uuid_t *uuid1, const ble_uuid_t *uuid2)
```

Comment:
Also, this change is small, but important: https://github.com/apache/mynewt-nimble/pull/193/files#diff-fa8c3d9c49d86b38eeeca3239b039642R94

It feels safer to compare things as unequal if you don't know how to compare them (cf. Java comparing things of incompatible types as unequal). If you are running without `BLE_HS_DEBUG`, returning zero by default will have surprising results downstream.